src/webgpu/idl/javascript.spec.ts

```diff
@@ -338,3 +355,110 @@ g.test('limits')
       );
     }
   });
+
+g.test('getter_replacement')
+  .desc(
+    `
+    Test that replacing getters on class prototypes works
+
+    This is a common pattern for shims and debugging libraries so make sure this pattern works.
+    `
+  )
+  .params(u => u.combine('type', kResources))
+  .fn(t => {
+    const { type } = t.params;
+    const { getters } = kResourceInfo[type];
+
+    const obj = createResource(t, type);
+    for (const getter of getters) {
+      // Check it's not 'ownProperty`
+      const properties = Object.getOwnPropertyDescriptor(obj, getter);
+      t.expect(
+        properties === undefined,
+        `Object.getOwnPropertyDescriptor(instance of ${type}, '${getter}') === undefined`
+      );
+
+      // Check it's actually a getter that returns a non-function value.
+      const origValue = (obj as unknown as Record<string, () => unknown>)[getter];
+      t.expect(typeof origValue !== 'function', `instance of ${type}.${getter} !== 'function'`);
+
+      // check replacing the getter on constructor works.
+      const ctorPrototype = obj.constructor.prototype;
+      const origProperties = Object.getOwnPropertyDescriptor(ctorPrototype, getter);
+      t.expect(
+        !!origProperties,
+        `Object.getOwnPropertyDescriptor(${type}, '${getter}') !== undefined`
+      );
+      try {
+        Object.defineProperty(ctorPrototype, getter, {
+          get() {
+            return 'testGetterValue';
+          },
+        });
+        const value = (obj as unknown as Record<string, string>)[getter];
+        t.expect(
+          value === 'testGetterValue',
+          `replacing getter: '${getter}' on ${type} returns test value`
+        );
+      } finally {
+        Object.defineProperty(ctorPrototype, getter, origProperties!);
```

Comment:
Line 388 should `assert` instead of `expect`, because it doesn't really make sense to continue the test if `getOwnPropertyDescriptor` didn't work.

This restoration code will fail if `origProperties` is undefined, leaving the page in a different state than it started.
This should also remove the need for `!` on `origProperties!`.